exe/task.cc

```diff
@@ -0,0 +1,192 @@
+                "properties": {
+                    "task": {
+                        "type": "string"
+                    },
+                    "input": {
+                        "type": "object"
+                    }
+                },
+                "required": [
+                    "task", "input"
+                ],
+                "type": "object",
+                "additionalProperties": true
+            },
+            "name": "run",
+            "results": {
+                "type": "object"
+            }
+        }
+    ],
+    "description": "Task runner module"
+}
+)";
+        return 0;
+    }
+
+    lth_log::setup_logging(boost::nowide::cerr);
+    lth_log::set_level(lth_log::log_level::error);
+
+    boost::nowide::cin >> std::noskipws;
+    istream_iterator<char> it(boost::nowide::cin), end;
+    string STDIN(it, end);
+    auto args = lth_jc::JsonContainer(STDIN);
+
+    auto taskname = args.get<string>({"input", "task"});
+    auto input = args.get<lth_jc::JsonContainer>({"input", "input"});
+
+    auto stdout_file = args.get<string>({"output_files", "stdout"});
+    auto stderr_file = args.get<string>({"output_files", "stderr"});
+    auto exitcode_file = args.get<string>({"output_files", "exitcode"});
+
+    auto exitcode = 255;
+    lth_jc::JsonContainer stdout_result;
+    string stderr_str;
+
+    string module, task = "init";
+    if (!lth_util::re_search(taskname, boost::regex("\\A(\\w+)\\z"), &module) &&
+        !lth_util::re_search(taskname, boost::regex("\\A(\\w+)::(\\w+)\\z"), &module, &task)) {
+        set_error(stdout_result, "invalid-task", _("Invalid task name '{1}'", taskname));
+    } else {
+        try {
+            auto filename = SYSTEM_PREFIX()+"/pxp-agent/tasks/"+module+"/tasks/"+task;
+
+            if (lth_exec::which(filename).empty()) {
+                set_error(stdout_result, "not-found", _("Task file '{1}' is not present or not executable", filename));
+            } else {
+                auto environment = generate_environment_from(input);
+                auto exec = lth_exec::execute(
+                    filename,
+                    {},  // arguments
+                    input.toString(),
+                    environment,
+                    0,  // timeout
+                    { lth_exec::execution_options::thread_safe,
+                      lth_exec::execution_options::merge_environment,
+                      lth_exec::execution_options::inherit_locale });
+
+                if (!isValidUTF8(exec.output)) {
+                    set_error(stdout_result, "output-encoding-error", _("Output cannot be represented as a JSON string"));
+                } else {
+                    stdout_result.set("output", exec.output);
+                }
+
+                stderr_str = exec.error;
+                exitcode = exec.exit_code;
+
+                if (exitcode != 0) {
+                    set_error(stdout_result, "task-error", _("The task errored with a code {1}", exitcode));
```

Comment:
Oh yeah. I had a question about parsing for _error somewhere as well. Since we probably won't parse for it here, generating an _error key is somewhat redundant.

Although currently, what I'm returning is `{"output": "...", "_error": "..."}` which means nothing specifically gets clobbered.